### PR TITLE
Add piper example

### PR DESCRIPTION
**Pull request by PeterD1524**
I am working on [LFX Workspace: Support piper as a new backend of the WASI-NN WasmEdge plugin #3483](https://www.github.com/WasmEdge/WasmEdge/issues/3483).
This PR adds a basic example of using the plugin to perform text-to-speech.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by hydai — APPROVED** [review]
LGTM. @dm4 PTAL

**Comment by dm4** [comment]
Could you please help add the following items? Thank you!

1. Provide a simple explanation of the related dependency installation in the `README.md`, such as `libespeak-ng`, `libonnxruntime`, `libpiper`, etc.
2. Add a GitHub workflow in `.github/workflows/piper.yml` to ensure that the example can run successfully.

File: wasmedge-piper/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "wasmedge-piper"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+serde_json = "1.0.118"
+wasmedge-wasi-nn = "0.7.1"
```
Comment (dm4) [inline wasmedge-piper/Cargo.toml:10]: We could update `wasmedge-wasi-nn` to `0.8.0`.

Comment (PeterD1524) [reply wasmedge-piper/Cargo.toml]: ok, https://github.com/second-state/WasmEdge-WASINN-examples/pull/145/commits/85cebe0efdd749c821fba248a1e75baf567021a7

**Comment by PeterD1524** [comment]
@dm4 I have added the following items.

> 1. Provide a simple explanation of the related dependency installation in the `README.md`, such as `libespeak-ng`, `libonnxruntime`, `libpiper`, etc.

https://github.com/second-state/WasmEdge-WASINN-examples/pull/145/commits/075409ef9447e294061d3eb42278061efec18a14

> 2. Add a GitHub workflow in `.github/workflows/piper.yml` to ensure that the example can run successfully.

https://github.com/second-state/WasmEdge-WASINN-examples/pull/145/commits/870e600286714e906a958511c6bbe71707b40233

File: .github/workflows/piper.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+      - name: Checkout WasmEdge
+        uses: actions/checkout@v4
+        with:
+          repository: WasmEdge/WasmEdge
+          path: WasmEdge
+
+      - name: Build WasmEdge with WASI-NN Piper plugin
+        run: |
+          cmake -GNinja -Bbuild -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_PLUGIN_WASI_NN_BACKEND=Piper
```
Comment (hydai) [inline .github/workflows/piper.yml:32]: We don't need to install `liblld-14-dev` here by setting the CMake option `-DWASMEDGE_USE_LLVM=OFF` to disable all AOT-related components.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]
LGTM. @dm4 PTAL.

File: .github/workflows/piper.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          sudo apt-get install ninja-build
+
+      - name: Install ONNX Runtime 1.14.1
+        run: |
+          curl -LO https://github.com/microsoft/onnxruntime/releases/download/v1.14.1/onnxruntime-linux-x64-1.14.1.tgz
+          tar zxf onnxruntime-linux-x64-1.14.1.tgz
+          sudo mv onnxruntime-linux-x64-1.14.1/include/* /usr/local/include/
+          sudo mv onnxruntime-linux-x64-1.14.1/lib/* /usr/local/lib/
+          rm -rf onnxruntime-linux-x64-1.14.1.tgz onnxruntime-linux-x64-1.14.1
+          ldconfig
```
Comment (hydai) [inline .github/workflows/piper.yml:31]: Since we are checking out the WasmEdge codebase, should we use the install script from the WasmEdge repo instead?

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/piper.yml (hunk 3 above)
Comment (PeterD1524) [reply .github/workflows/piper.yml]: ok https://github.com/second-state/WasmEdge-WASINN-examples/pull/145/commits/c6fd2ba2b0b529c21f9a066c911509142f182b29

**Review by hydai — APPROVED** [review]

**Pull request merged**